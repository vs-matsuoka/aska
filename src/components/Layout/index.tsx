import Image from 'next/future/image';
import MenuBar from 'components/MenuBar';

type LayoutProps = {
  children?: React.ReactNode;
};

function Layout({ children }: LayoutProps) {
  return (
    <div
      className="grid min-h-screen w-full justify-center"
      style={{
        gridTemplateRows: 'auto 1fr',
        gridTemplateColumns: '100%'
      }}
    >
      <div className="fixed">
        <div className="h-screen w-screen"></div>
        <Image
          src="/Entry/21_Entry_pic_BG.png"
          alt="background"
          className="-z-10"
          fill
          style={{
            objectFit: 'cover'
          }}
        />
      </div>
      <div className="fixed w-full">
        <MenuBar />
      </div>
      <div className="pt-20"></div>
      <div className="relative flex items-center justify-center">
        {children}
      </div>
    </div>
  );
}

export default Layout;
