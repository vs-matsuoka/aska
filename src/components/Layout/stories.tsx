import { Story, Meta } from '@storybook/react';
import Layout from '.';

export default {
  title: 'Layout',
  component: Layout
} as Meta;

export const Default: Story = () => <Layout></Layout>;
