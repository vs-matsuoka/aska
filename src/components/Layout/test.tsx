import { render, screen, waitFor, within } from '@testing-library/react';

import Layout from '.';

jest.mock('next/router', () => ({ useRouter: jest.fn() }));

describe('<Layout />', () => {
  it('正しくレンダリングされる', async () => {
    render(
      <Layout>
        <h1>AAAAAAAAAAAAAA</h1>
      </Layout>
    );

    const { getByText } = within(screen.getByTestId('children'));

    expect(getByText('AAAAAAAAAAAAAA')).toBeInTheDocument();
  });
});
