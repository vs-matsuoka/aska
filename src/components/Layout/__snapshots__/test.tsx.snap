// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> should render the heading 1`] = `
<h1>
  Layout
</h1>
`;
